pkg/file_system/status_writer.go

```diff
@@ -0,0 +1,59 @@
+package file_system
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+type IStatusWriter interface {
+	Write() error
+}
+
+type StatusWriter struct {
+	file    string
+	message []byte
+	perm    os.FileMode
+}
+
+func NewStatusWriter(file string, message []byte) StatusWriter {
+	return StatusWriter{
+		file:    file,
+		message: message,
+		perm:    0644,
+	}
+}
+func (w StatusWriter) Write() error {
+	return ioutil.WriteFile(w.file, w.message, w.perm)
+}
+
+type StatusAppender struct {
+	file      string
+	message   []byte
+	perm      os.FileMode
+	fileFlags int
+}
+
+func NewStatusAppender(file string, message []byte) StatusAppender {
+	return StatusAppender{
+		file:      file,
+		message:   message,
+		perm:      0644,
+		fileFlags: os.O_APPEND | os.O_CREATE | os.O_WRONLY,
+	}
+}
+func (a StatusAppender) Write() error {
+	file, openErr := os.OpenFile(a.file, a.fileFlags, a.perm)
+	if openErr != nil {
+		return openErr
+	}
+
+	_, writeErr := file.Write(a.message)
+	if writeErr != nil {
+		closeErr := file.Close()
+		if closeErr != nil {
+			return closeErr
+		}
+		return writeErr
+	}
+	return file.Close()
```

Comment:
Would it be safer to use `defer` here? I'm not positive you could return that error in the event close fails, but that seems like the classic use case for it.

Reply:
Hm, I'm not sure the best way to handle this. I was basing this on how golang handles it in [`ioutil.WriteFile`](https://github.com/golang/go/blob/888a0c8ef6afb752aafd147eda40d62796d87cb3/src/io/ioutil/ioutil.go#L79) (which uses the `os` package as well. 

I guess I would expect that returning `file.Close` the error would be returned if there was one.

Reply:
Fair enough although I'd argue your code is more readable :)